src/wp-includes/class-wp-query.php

```diff
@@ -3292,8 +3292,21 @@ public function get_posts() {
 			 *
 			 * @param bool     $split_the_query Whether or not to split the query.
 			 * @param WP_Query $query           The WP_Query instance.
+			 * @param string   $old_request     The complete SQL query before filtering.
+			 * @param string[] $clauses {
+			 *      Associative array of the clauses for the query.
+			 *
+			 *      @type string $where    The WHERE clause of the query.
+			 *      @type string $groupby  The GROUP BY clause of the query.
+			 *      @type string $join     The JOIN clause of the query.
+			 *      @type string $orderby  The ORDER BY clause of the query.
+			 *      @type string $distinct The DISTINCT clause of the query.
+			 *      @type string $fields   The SELECT clause of the query.
+			 *      @type string $limits   The LIMIT clause of the query.
+			 * }
+			 * @param array    $q Defined query variables.
```

Comment:
Thsi looks available at `$query->query_vars`

Reply:
$q is not the same $query->query_vars. The $q variable is change here. 

https://github.com/WordPress/wordpress-develop/blob/ea502c457090896960109f750347e265c052b276/src/wp-includes/class-wp-query.php#L1923-L2032

Reply:
$q is a reference 

https://github.com/WordPress/wordpress-develop/blob/ea502c457090896960109f750347e265c052b276/src/wp-includes/class-wp-query.php#L1884